Upgrade to latest prodcon build and add targeting packs for net46 support

crummel opened the pull request:
Testing build with net46 changes

crummel commented:
@dotnet-bot test this please

crummel commented:
CI seems messed up, closing and reopening.

crummel commented:
@dseefeld @dagood This should fix the net46 issues for us for now.

dagood commented on patches/msbuild/0001-Add-TargetingPack-and-enable-net46-builds-for-source.patch:
```diff
@@ -0,0 +1,62 @@
++
++  <Target Name="CopyMscorlib" AfterTargets="Build">
++    <Copy SourceFiles="$(NuGetPackageRoot)\netstandard.library\2.0.0\build\netstandard2.0\ref\netstandard.dll" DestinationFolder="$(ArtifactsBinDir)/MSBuild/$(TargetFramework)/ref" />
++    <Copy SourceFiles="$(NuGetPackageRoot)\netstandard.library\2.0.0\build\netstandard2.0\ref\mscorlib.dll" DestinationFolder="$(ArtifactsBinDir)/MSBuild/$(TargetFramework)/ref" />
```
What is this change about? I don't see it on https://github.com/NuGet/NuGet.Client/pull/2287.

crummel replied:
It's actually on the MSBuild PR: https://github.com/Microsoft/msbuild/pull/3389/files#diff-4d3fc7780500131817fceb95ddcea3feR983.  Copying mscorlib after the build means it doesn't get pulled into the references, it's a workaround for now.

crummel commented:
Ah, I rebased your commit and got my patch commits in there as well.  I changed them back in my commits  so I'll squash when this gets merged so it doesn't show up as changing them and then reverting.  Thanks for the catch.

dagood commented:
@dotnet-bot test OSX10.12 Release (java.nio.channels.ClosedChannelException)

dagood commented:
@dotnet-bot test RHEL7.2 Tarball Release (I didn't see anything queued on https://ci.dot.net/job/dotnet_source-build/job/release_2.1/job/RHEL7.2_Tarball_Release/ ...)

dagood commented:
Realized that the update misses the AspNetCore version and ProdCon feed updates. Filed https://github.com/dotnet/source-build/issues/608 to fix.

dagood commented:
I'm working on batching up the rest of the 2.1.1 changes into a branch based on this PR so I can open the PR for those after this one gets in.

dagood commented:
@dotnet-bot test OSX10.12 Release (java.nio.channels.ClosedChannelException)

dagood commented:
@dotnet-bot test OSX10.12 Release (last try didn't seem to do anything.)
